src/parsec/agent/runners/lua/impl.cpp

```diff
@@ -138,25 +138,51 @@ namespace cbdc::parsec::agent::runner {
         return buf;
     }
 
+    auto lua_runner::get_stack_integer(int index) -> std::optional<int64_t> {
+        if(lua_isinteger(m_state.get(), index) != 1) {
+            return std::nullopt;
+        }
+        return lua_tointeger(m_state.get(), index);
+    }
+
     void lua_runner::schedule_contract() {
         int n_results{};
         auto resume_ret = lua_resume(m_state.get(), nullptr, 1, &n_results);
         if(resume_ret == LUA_YIELD) {
-            if(n_results != 1) {
-                m_log->error("Contract yielded more than one key");
+            if(n_results > 2) {
+                m_log->error("Contract yielded more than two keys");
                 m_result_callback(error_code::yield_count);
                 return;
             }
+
+            auto lock_level = broker::lock_type::write;
+            if(n_results == 2) {
```

Comment:
also it might make sense to add a check here to handle the case where  the call to `lua_resume` does not yield any results (unless it is the expectation that this will be handled in the contract specification)